Userland/Libraries/LibGfx/BMPLoader.cpp

```diff
@@ -744,35 +746,35 @@ static bool decode_bmp_v5_dib(BMPLoadingContext& context, InputStreamer& streame
     return true;
 }
 
-static bool decode_bmp_dib(BMPLoadingContext& context)
+static ErrorOr<void> decode_bmp_dib(BMPLoadingContext& context)
 {
     if (context.state == BMPLoadingContext::State::Error)
-        return false;
+        return Error::from_string_literal("Error before starting decode_bmp_dib");
 
     if (context.state >= BMPLoadingContext::State::DIBDecoded)
-        return true;
+        return {};
 
-    if (!context.is_included_in_ico && context.state < BMPLoadingContext::State::HeaderDecoded && !decode_bmp_header(context))
-        return false;
+    if (!context.is_included_in_ico && context.state < BMPLoadingContext::State::HeaderDecoded && decode_bmp_header(context).is_error())
+        return Error::from_string_literal("Error decoding bmp header");
 
     u8 header_size = context.is_included_in_ico ? 0 : bmp_header_size;
 
     if (!context.is_included_in_ico && context.file_size < (u8)(header_size + 4))
-        return false;
+        return Error::from_string_literal("file size too short");
```

Comment:
Capitalize the first word (File) and let's use a more descriptive error. Something like "File is smaller than expected header size".